File: src/grader/ex00e5/consecutive_ones.rb
def gen(sol, l, c, mc, n, k)
  if (l == n)
    if (mc >= k)
      puts sol
    end
    return
  end
  gen(sol + "0", l + 1, 0, mc, n, k)
  gen(sol + "1", l + 1, c + 1, mc > c ? mc : c + 1, n, k)
end

n, k = gets.split.map { |x| x.to_i }
gen("", 0, 0, 0, n, k)

File: src/grader/ex00e5/consecutive_ones.py
def r(s, l, c, d):
    if c == k:
        d = True
    if l == n:
        if d:
            print(s)
    else:
        r(s+"0", l+1, 0, d)
        r(s+"1", l+1, c+1, d)


n, k = (int(x) for x in input().split())
r("", 0, 0, False)
